crates/subspace-solving/src/codec.rs

```diff
@@ -133,16 +134,30 @@ impl SubspaceCodec {
 
         #[cfg(feature = "cuda")]
         if self.cuda_available {
-            let pieces_to_process = pieces.len() / PIECE_SIZE / GPU_PIECE_BLOCK * GPU_PIECE_BLOCK;
-
-            self.batch_encode_cuda(
-                &mut pieces[..pieces_to_process * PIECE_SIZE],
-                &piece_indexes[..pieces_to_process],
-            )?;
-
-            // Leave the rest for CPU
-            pieces = &mut pieces[pieces_to_process * PIECE_SIZE..];
-            piece_indexes = &piece_indexes[pieces_to_process..];
+            let mut pieces_to_process = pieces.len() / PIECE_SIZE;
+
+            // GPU will accept multiples of 1024 pieces
+            if pieces_to_process >= 1024 {
+                pieces_to_process = pieces_to_process / GPU_PIECE_BLOCK * GPU_PIECE_BLOCK;
```

Comment:
It looks like noop at first glance. This should be more obvious.
```suggestion
                pieces_to_process -= pieces_to_process % GPU_PIECE_BLOCK;
```

Reply:
I actually wrote it that way. But Nazar wanted to change it to the current version. I believe the reasoning may be some compiler optimization (dividing by a power of 2 can be optimized). 
I also prefer using a modulo operator btw, more high-level and readable

Reply:
It is integer arithmetic, it is not noop unless division is done without remainder. You version, Ivan, looks fancy, but I'm not sure how much more readable it is. In practice I wouldn't be surprised if it compiles to similar if not the same code.